extensions/pkg/controller/backupbucket/backupbucket_test.go

```diff
@@ -0,0 +1,337 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backupbucket_test
+
+import (
+	"context"
+	"time"
+
+	backupbucketcontroller "github.com/gardener/gardener/extensions/pkg/controller/backupbucket"
+	testactuator "github.com/gardener/gardener/extensions/pkg/controller/backupbucket/test"
+	"github.com/gardener/gardener/extensions/pkg/controller/test"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/test/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	gomegatypes "github.com/onsi/gomega/types"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var _ = Describe("BackupBucket", func() {
+	AfterEach(func() {
+		framework.RunCleanupActions()
+	})
+
+	It("should successfully create and delete a BackupBucket (ignoring operation annotation)", func() {
+		prepareAndRunTest(true)
+	})
+
+	It("should successfully create and delete a BackupBucket (respecting operation annotation)", func() {
+		prepareAndRunTest(false)
+	})
+})
+
+func prepareAndRunTest(ignoreOperationAnnotation bool) {
+	By("setup and start manager")
+	Expect(createAndStartManager(ignoreOperationAnnotation)).NotTo(HaveOccurred())
+
+	By("setup client for test")
+	c, err := client.New(restConfig, client.Options{Scheme: kubernetes.SeedScheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("generate namespace name for test")
+	namespace, err := generateNamespaceName()
+	Expect(err).NotTo(HaveOccurred())
+
+	runTest(c, namespace, ignoreOperationAnnotation)
+}
+
+func createAndStartManager(ignoreOperationAnnotation bool) error {
+	mgrContext, mgrCancel := context.WithCancel(ctx)
+
+	var cleanupHandle framework.CleanupActionHandle
+	cleanupHandle = framework.AddCleanupAction(func() {
+		defer func() {
+			By("stopping manager")
+			mgrCancel()
+		}()
+
+		framework.RemoveCleanupAction(cleanupHandle)
+	})
+
+	mgr, err := manager.New(restConfig, manager.Options{MetricsBindAddress: "0"})
+	if err != nil {
+		return err
+	}
+
+	if err := extensionsv1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
+		return err
+	}
+
+	if err := testactuator.AddToManagerWithOptions(mgr, ignoreOperationAnnotation); err != nil {
+		return err
+	}
+
+	go func() {
+		if err := mgr.Start(mgrContext.Done()); err != nil {
+			panic(err)
+		}
```

Comment:
Can you handle this gracefully with a matcher?
```suggestion
		GinkgoRecover()
		if err := mgr.Start(mgrContext.Done()); err != nil {
			Expect(err).NotTo(HaveOccurred())
		}
```